pkg/runtime/runtime.go

```diff
@@ -2191,6 +2241,9 @@ func (a *DaprRuntime) Shutdown(duration time.Duration) {
 			log.Warnf("error closing API: %v", err)
 		}
 	}
+	if a.tracerProvider != nil {
```

Comment:
This needs to go into a goroutine or it can delay the shutdown.


```go
shutdownCtx, shutdownCancel := context.WithCancel(context.Background())
go func() {
	if a.tracerProvider != nil {
		a.tracerProvider.Shutdown(shutdownCtx)
	}
}()

log.Infof("Waiting %s to finish outstanding operations", duration)
<-time.After(duration)
shutdownCancel()
```